scala-style-guide/README.md

```diff
@@ -0,0 +1,1287 @@
+* Не используйте в импорте оператор `_`, если вы не импортируете более 4 сущностей. При таком импорте код становится менее устойчивым к внешним воздействиям, происходит лишнее загромождение пространоства имен, увеличивается риск появления конфликта имен. Более подробно можно почитать [тут](https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad).
+* Всегда импортируйте пакеты используя полный путь (например, `scala.util.Random`) вместо относительных (например, `util.Random`).
+* Импорт сортируется в следущем порядке:
+   * `java.*` и  `javax.*`
+   * `scala.*`
+   * Сторонние библиотеки (`org.*`, `com.*` и т.д.).
+* Конфигурация стиля кода InteliJ позволяет автоматически группировать импорт по пространству имен. Используйте шаблон import layout в Intelij, для этого зайдите в настройки `Settings->Editor->Code Style->Scala`, затем выберете вкладку Imports и приведите import layout к следующему виду:
+
+	```scala
+	java
+	javax
+	_______ blank line _______
+	scala
+	_______ blank line _______
+	akka
+	all other imports
+	```
+* Вы также должны регулярно запускать Intellij'S Optimize Imports (`Code->Optimize Imports`) перед созданием пулреквеста, для поддержания чистоты `import`.
+
+### <a name='class_ordering'>Сортировка внутри класса</a>
+Сортировка внутри класса должна идти по порядку по следующему шаблону: 	
+	
+* импорты внутри класса(если таковые имеются)
+* переменные
+* методы
+  * методы поиска
+  * методы создания
+  * методы обновления и/или изменения
+  * методы удаления
+   
+   ```scala
+	class CampaignController {
+	  //сначала импорты
+	  import Helper._
+	  ...
+	  
+	  //потом переменные
+	  val PORT = 3000
+	  ...
+	  
+	  //затем методы
+	  
+	  //методы поиска
+	  def all = ...
+	  ...
+	  
+	  //методы создания
+	  def create = ...
+	  ...
+
+     //методы обновления и/или изменения
+	  def update = ...
+	  
+	  def copy = ...
+	  ...
+	    
+	  //методы удаления
+	  def delete = ...
+	  ...
+	}
+   ```
+
+### <a name='pattern_matching'>Pattern Matching</a>
+* Для метода, все тело которого является выражением, поместите match на ту же строку что и объявление метода.
+
+   ```scala
+	def title(status: String) = status match {
+	...
+	}
+	```
+	
+* Если при вызове функции существует только один case, то поместите его в ту же строку, что и вызов функции.
+
+   ```scala
+	statuses.zipWithIndex.map { case (status, index) =>
+	...
+	}
+	```
+	
+* Если при вызове функции существует несколько case, то поместите каждый case на новую строку.
+  
+   ```scala
+	list.map { 
+	  case a: User => ...
+	  case b: Hall => ...
+	}
+	```
+	
+* Если тело case помещается в одну строку, то так и делайте, иначе оберните его в фигруные скобки.
+
+	```scala
+	//хорошо
+	status match { 
+	  case ACTIVE => "Активный"
+	  case PAUSED => "Пауза"
+	}
+	
+	//хорошо
+	status match { 
+	  case ACTIVE => {
```

Comment:
тут наверное стоит опираться на количество кода внутри case. Хотя я никогда скобки внутри не юзал

Reply:
> Почему? Мне вот например неудобно смотреть на case без `{}`, особенно на довольно большие при формировании экселек или в том же receive актора. А так редактор сразу показывает где начало и конец у case и ревьювить легче. Я просто не раз сталкивался на ревью с таким и реально невозможно это ревьювить, больно.

Если у тебя большие формирования экселек, то их надо переносить в отдельные функции.
IDEA подсказывает что скобки после case лишние, а IDEA плохого не посоветует :)

Reply:
IDEA много чего подсказывает, но далеко не все из этого верно. Если сейчас дать добро на case без {}, то я вангую что будут большие case, которые читать не возможно будет. Можем проголосовать если хотите.

Reply:
> будут большие case
тогда надо добавить в гайд инфу о больших case

Always omit braces in case clauses - Согласно официальному [scala style guide](https://docs.scala-lang.org/style/control-structures.html#curly-braces)